### PR TITLE
Add xperimental features and add table for how sync works

### LINKED ISSUE #1626: WIP: Add experimental `--Xsnapsync-bft-enabled` option
_This is still being developed. Check[ Eng PR](https://github.com/hyperledger/besu/pull/7140) to get status on feature_

* Update docs to reflect "how do the different sync modes work" and add private syncing to https://besu.hyperledger.org/development/public-networks/get-started/connect/sync-node

* Document `--Xsnapsync-bft-enabled` which enables snap sync for BFT chains.

**Related to**:  https://github.com/hyperledger/besu/pull/7140/commits

### PR DESCRIPTION
**Pull request by joaniefromtheblock**
## Description

-  Update explanation of syncs and add experimental feature

### Issue(s) fixed

Fixes #1626 

### Preview

- https://besu-docs-git-xperimental-method-update-hyperledger.vercel.app/public-networks/get-started/connect/sync-node
- https://besu-docs-git-xperimental-method-update-hyperledger.vercel.app/private-networks/reference/cli/options

### CODE REVIEW

**Comment by joaniefromtheblock** [comment]
@matthew1001 Please review when you have a chance.

File: docs/public-networks/get-started/connect/sync-node.md
Comment (siladu) [inline docs/public-networks/get-started/connect/sync-node.md:59]: I'm not sure whether this section is adding any extra information that isn't already documented elsewhere. What is the intent with this section?

Maybe we need to call out any configuration combinations that don't work for private networks - I think they all will eventually but maybe not yet - I'll defer to @matthew1001 for that.

File: docs/public-networks/get-started/connect/sync-node.md
Comment (siladu) [inline docs/public-networks/get-started/connect/sync-node.md:274]: might be worth mentioning this builds an archive node, with full state history

File: docs/public-networks/get-started/connect/sync-node.md
Comment (siladu) [inline docs/public-networks/get-started/connect/sync-node.md:271]: There's two types of data being synced: chain data (blocks), and world state data.
Snap/checkpoint sync chain data the same way as fast sync. The main difference is how they handle world state data.

File: docs/public-networks/get-started/connect/sync-node.md
Comment (joaniefromtheblock) [reply docs/public-networks/get-started/connect/sync-node.md:59]: Thanks for the review @siladu . The table was intended to provide a summary or easily digestible overview of the different types of synchronization. While I agree that much of the material is repetitive, I also think  it's a bit difficult for users to understand the comparison between these types. Do you think we should remove the table entirely, or should we rework it to provide more value?

Comment (siladu) [reply docs/public-networks/get-started/connect/sync-node.md:59]: I think the table is good, I'm referring to this highlighted preamble section

**Comment by joaniefromtheblock** [comment]
@matthew1001 please take another look. Does this PR look like it is good to go now?

**Comment by joaniefromtheblock** [comment]
Thank you so much! Will add suggestions

**Review by bgravenorst — COMMENTED** [review]
Some initial feedback. Will circle back once implemented.

File: docs/public-networks/concepts/sync-node.md
Comment (siladu) [reply docs/public-networks/concepts/sync-node.md:52]:
```suggestion
| Snap | Default for Mainnet. Efficient sync starting from the genesis block, the successor to Fast sync. | Besu version 22.4.0 or later | Downloads as many leaves of the trie as possible, reconstructs the trie locally. Snap is faster than Fast sync. | Cannot switch from fast sync to snap sync mid-process. |
```

File: docs/public-networks/concepts/sync-node.md
Comment (siladu) [inline docs/public-networks/concepts/sync-node.md:59]:
```suggestion
Ethereum nodes sync two types of data: chain data (blocks) and world state data (account storage). 
```

**Review by siladu — COMMENTED** [review]
Getting close, few more tweaks IMO

File: docs/private-networks/concepts/private-network-syncing.md
Comment (joaniefromtheblock) [reply docs/private-networks/concepts/private-network-syncing.md:37]: Tried using this and it looks eve more confusing :(

File: docs/private-networks/concepts/private-network-syncing.md
Comment (joaniefromtheblock) [inline docs/private-networks/concepts/private-network-syncing.md:20]:
```suggestion
When [setting up a private network](../tutorials/privacy.md), consider the following:
```

File: docs/public-networks/concepts/sync-node.md
Comment (joaniefromtheblock) [inline docs/public-networks/concepts/sync-node.md:2]:
```suggestion
title: Public network syncing
```

File: docs/public-networks/concepts/sync-node.md
Comment (alexandratran) [inline docs/public-networks/concepts/sync-node.md:55]: There seems to be some overlap across the "Method" and "Description" columns. Since this is just an overview table, I suggest making this it smaller by combining the two (maybe just "Description"). You can specify which modes are defaults for which networks in their respective extended sections.

File: docs/public-networks/concepts/sync-node.md
Comment (alexandratran) [inline docs/public-networks/concepts/sync-node.md:9]: After reviewing this page, I'm wondering if we should take this opportunity to restructure it more intuitively for a "Concepts" page. My suggested restructure:

- Node synchronization
  - Node types
    - Full nodes
    - Archive nodes
  - Sync modes
    - Snap sync
    - Checkpoint sync
    - Fast sync
    - Full sync
  - Sync times
  - Storage

File: docs/private-networks/concepts/private-network-syncing.md
Comment (alexandratran) [inline docs/private-networks/concepts/private-network-syncing.md:9]:
```suggestion
sidebar_label: Node synchronization
sidebar_position: 4
description: Learn about node synchronization for private networks.
tags:
  - private networks
---

# Node synchronization for private networks
```

For more consistency with my suggestion on the public networks doc

File: docs/private-networks/concepts/private-network-syncing.md
Comment (alexandratran) [inline docs/private-networks/concepts/private-network-syncing.md:25]:
```suggestion
For private, permissioned blockchain networks, Besu uses the same [synchronization 
modes](/public-networks/concepts/sync-node) as public networks, but with specific configurations for private network needs.

To sync Besu on a private network:

- Ensure all nodes use compatible sync modes and configurations.
- Configure the network with a custom genesis file.
- Set the network ID and bootnodes specific to your private network.
- Implement permissioning features to control network access.
```

Suggestion for conciseness.

File: docs/private-networks/concepts/private-network-syncing.md
Comment (alexandratran) [inline docs/private-networks/concepts/private-network-syncing.md:47]: Similar comment to the public network table, I think we can simplify and combine "Method" and "Description" into one column. Also add links to the relevant sync mode sections for each mode (snap, checkpoint, etc). There should also be consistency across this page and the public networks page on what location we recommend snap sync (in the table or in a note).

**Review by alexandratran — COMMENTED** [review]
Nice work so far. Added my first pass of comments.

File: docs/public-networks/concepts/sync-node.md
Comment (joaniefromtheblock) [reply docs/public-networks/concepts/sync-node.md]: May we open another PR for this? I have had the PR open for while and I think this promotes a bit of scope creep.

File: docs/public-networks/concepts/node-sync.md
Comment (alexandratran) [inline docs/public-networks/concepts/node-sync.md:40]:
```suggestion
| [Snap](#snap-synchronization) | Default for Mainnet. Efficient sync from genesis block, downloading as many trie leaves as possible and reconstructing locally. Faster than fast sync. | Besu version 22.4.0 or later | Cannot switch from fast sync to snap sync mid-process. |
```

File: docs/public-networks/concepts/node-sync.md
Comment (alexandratran) [inline docs/public-networks/concepts/node-sync.md:42]:
```suggestion
| [Fast](#fast-synchronization) | Default for named networks except `dev`. Downloads block headers and transaction receipts, verifies chain from genesis block. | None | Might become impossible to fast sync Ethereum Mainnet in the future. |
```

File: docs/public-networks/concepts/node-sync.md
Comment (alexandratran) [inline docs/public-networks/concepts/node-sync.md:54]:
```suggestion
Snap and checkpoint syncs handle chain data similarly to fast sync, but differ in how they process world state data.
```

File: docs/public-networks/concepts/node-sync.md
Comment (alexandratran) [inline docs/public-networks/concepts/node-sync.md:61]:
```suggestion
See [Node synchronization for private networks](../../private-networks/concepts/node-sync-private.md) for more information.
```

File: docs/private-networks/concepts/node-sync-private.md
Comment (alexandratran) [inline docs/private-networks/concepts/node-sync-private.md:30]:
```suggestion
`--Xsnapsync-bft-enabled` is an early access feature available in Besu version 24.7.1 and later.
It is not stable, so use this option with caution.

Use `--Xsnapsync-bft-enabled` with `--sync-mode=SNAP` to enable snap sync in QBFT and IBFT 2.0 private networks.
```

Suggested simplification. Same applies to the other copies of the note.

Comment (alexandratran) [reply docs/private-networks/concepts/node-sync-private.md:30]: Also, maybe move this note after the table, above the note that recommends snap sync?

File: docs/private-networks/concepts/node-sync-private.md
Comment (alexandratran) [inline docs/private-networks/concepts/node-sync-private.md:38]:
```suggestion
| [Snap](../../public-networks/concepts/node-sync.md#snap-synchronization) | Recommended for fastest sync and lowest storage requirements on Mainnet. Downloads as many leaves of the trie as possible, reconstructs the trie locally. Faster than fast sync. | Available as an _early access feature_ in Besu version 24.7.1 or later | Cannot switch from fast sync to snap sync mid-process. |
```

File: docs/private-networks/concepts/node-sync-private.md
Comment (alexandratran) [inline docs/private-networks/concepts/node-sync-private.md:36]: Can you copyedit all the descriptions on this private network table to ensure they are the same as the public network table where applicable? I see some wording differences that can be fixed.

File: docs/public-networks/concepts/sync-node.md
Comment (alexandratran) [reply docs/public-networks/concepts/sync-node.md]: That's fine with me 👍

**Review by alexandratran — APPROVED** [review]
LGTM once final notes are addressed.

**Pull request merged**
